src/interface_adapter/Login/loginState.java

```diff
@@ -1,6 +1,30 @@
 package interface_adapter.Login;
 
 public class loginState {
-    public static void setUsername(String username) {
+
+    private String username = "";
+    private String password = "";
+    private String error = null;
+
+    public void setUsername(String user) {
+        this.username = user;
+    }
+
+    public void setError(String uidState) {
+        this.error = null;
+    }
```

Comment:
A bit worried about this implementation, we might need to come back to this later.